### PR TITLE
Config

### PR DESCRIPTION
**Pull request by Kasperekx**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,9 @@
 {
   "singleQuote": true,
-  "trailingComma": "all"
+  "trailingComma": "all",
```
Comment (mcz90) [inline .prettierrc:3]: That would be ideal to have the same settings in all TypeScript projects.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: '3.8'
+services:
+  dev-db:
```
Comment (mcz90) [inline docker-compose.yml:3]: why not just "development"?  Do you thing we are going to work on stage and prod only or development, stage and prod ?

File: package.json [hunk 3]
```diff
@@ -17,14 +17,22 @@
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json"
+    "test:e2e": "jest --config ./test/jest-e2e.json",
+    "postinstall": "husky install",
+    "prepare": "husky install"
   },
   "dependencies": {
```
Comment (mcz90) [inline package.json:24]: Can we have any error or at least warning during build when any dependency can be updated to newer, stable version?

File: src/app.module.ts [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
-import { AppController } from './app.controller';
-import { AppService } from './app.service';
+import { DatabaseModule } from './infrastructure/config/database/database.module';
+import { ApplicationModule } from './application/application.module';
 
 @Module({
-  imports: [],
-  controllers: [AppController],
-  providers: [AppService],
+  imports: [DatabaseModule, ApplicationModule],
```
Comment (mcz90) [inline src/app.module.ts:6]: App and Application namings are too similar.
How about
App->Application 
Application->Core
?

First can stay as App, but I think Core will be more reliable than Application.

File: src/application/controllers/index.ts
Comment (mcz90) [inline src/application/controllers/index.ts:1]: Are those all empty files needed?

File: src/infrastructure/config/database/database.module.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { EnvModule } from '../env/env.module';
+import { DatabaseService } from './database.service';
+
+@Module({
+  imports: [
+    TypeOrmModule.forRootAsync({
+      imports: [EnvModule],
```
Comment (mcz90) [inline src/infrastructure/config/database/database.module.ts:9]: Env -> Environment?
Shortcuts are not always useful

File: src/infrastructure/config/env/env.service.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { parse } from 'dotenv';
+import { readFileSync } from 'fs';
+
+export class EnvService {
+  private readonly config: { [key: string]: string | undefined } = {};
```
Comment (mcz90) [inline src/infrastructure/config/env/env.service.ts:5]: Why this can be undefined?

File: src/infrastructure/config/env/env.service.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  constructor(path: string) {
+    this.config = parse(readFileSync(path));
+  }
+
+  get(key: string): string {
```
Comment (mcz90) [inline src/infrastructure/config/env/env.service.ts:10]: Why returning empty string? If something doesn't exist it's more readable to return null in my opinion. Imagine the situation of adding optional field to database, is it petter to get null or empty string to know if this should be displayed?

File: src/infrastructure/config/database/database.module.ts (hunk 5 above)
Comment (Kasperekx) [reply src/infrastructure/config/database/database.module.ts]: Done

File: src/infrastructure/config/env/env.service.ts (hunk 6 above)
Comment (Kasperekx) [reply src/infrastructure/config/env/env.service.ts]: I change to null, will be more logic

**Pull request merged**